pkg/lib/cockpit.d.ts

```diff
@@ -0,0 +1,42 @@
+declare module 'cockpit';
+
+type JsonValue = null | boolean | number | string | JsonObject | Array<JsonValue>;
+type JsonObject = { [name: string]: JsonValue };
+
+export interface EventMixin {
+    addEventListener(event: string, callback: any);
+    removeEventListener(event: string, callback: any);
+}
+
+function addEventListener(event: string, callback: any);
+function removeEventListener(event: string, callback: any);
+
+class BasicError {
+    problem: string;
+    message: string;
+    toString(): string;
+}
+
+const location : {
+    url_root: string;
+    options: Array<string>;
```

Comment:
Isn't this a dict?

```
pkg/systemd/logs.jsx:252:85 - error TS2339: Property 'boot' does not exist on type 'string[]'.

252                                                 isDisabled={options.boot && options.boot !== "0"}
                                                                                        ~~~~
```

Reply:
Right.  It's the path that's an `Array<string>`.  Thanks!